core/vm/logger_trace.go

```diff
@@ -47,10 +47,16 @@ func traceLastNAddressCode(n int) traceFunc {
 
 func traceCodeWithAddress(l *StructLogger, address common.Address) {
 	code := l.env.StateDB.GetCode(address)
+
 	keccakCodeHash := l.env.StateDB.GetKeccakCodeHash(address)
-	poseidonCodeHash := l.env.StateDB.GetPoseidonCodeHash(address)
+	codeHash := keccakCodeHash
+	poseidonCodeHash := common.Hash{}
+	if !l.env.chainRules.IsEuclid {
+		poseidonCodeHash = l.env.StateDB.GetPoseidonCodeHash(address)
```

Comment:
So what will happen on l2geth-mpt nodes pre-euclid?

Reply:
They will panic, but it is in the RPC handler.

Reply:
fixed in https://github.com/scroll-tech/go-ethereum/pull/1119/commits/51f3ab9ade8d987a733be45f6700db047fd12df7

Reply:
nice catch.